server/game/cards/01_SOR/IdenVersioInfernoSquadCommander.ts

```diff
@@ -0,0 +1,51 @@
+import AbilityHelper from '../../AbilityHelper';
+import { LeaderUnitCard } from '../../core/card/LeaderUnitCard';
+import { Aspect, EventName, RelativePlayer } from '../../core/Constants';
+import { StateWatcherRegistrar } from '../../core/stateWatcher/StateWatcherRegistrar';
+import { UnitsDefeatedThisPhaseWatcher } from '../../stateWatchers/UnitsDefeatedThisPhaseWatcher';
+
+export default class IdenVersioInfernoSquadCommander extends LeaderUnitCard {
+    private unitsDefeatedThisPhaseWatcher: UnitsDefeatedThisPhaseWatcher;
+
+    protected override getImplementationId() {
+        return {
+            id: '2048866729',
+            internalName: 'iden-versio#inferno-squad-commander',
+        };
+    }
+
+    protected override setupStateWatchers(registrar: StateWatcherRegistrar): void {
+        this.unitsDefeatedThisPhaseWatcher = AbilityHelper.stateWatchers.unitsDefeatedThisPhase(registrar, this);
+    }
+
+    protected override setupLeaderSideAbilities() {
+        this.addActionAbility({
+            title: 'Heal 1 from base if an opponent\'s unit was defeated this phase',
+            cost: AbilityHelper.costs.exhaustSelf(),
+            immediateEffect: AbilityHelper.immediateEffects.conditional({
+                condition: (context) => {
+                    const opponentUnitsDefeatedThisPhase = this.unitsDefeatedThisPhaseWatcher.getDefeatedUnitsControlledByPlayer(context.source.controller.opponent);
+                    return opponentUnitsDefeatedThisPhase.length > 0;
```

Comment:
Will opponentUnitsDefeatedThisPhase always be an array or could it ever be undefined/null?